torch/nn/modules/sparse.py

```diff
@@ -50,6 +50,24 @@ class Embedding(Module):
         output. The gradient for this vector from :class:`~torch.nn.Embedding`
         is always zero.
 
+    .. note::
+        When :attr:`max_norm` is not ``None``, :class:`Embedding`'s forward method will modify the
+        :attr:`weight` Tensor in-place, rather than setting ``Embedding.weight`` to a new Tensor.
+        Because of this, if you want to perform some calculation with ``Embedding.weight`` before
```

Comment:
Can we be clearer about the reason for this? Maybe something like, 

"Since tensors needed for gradient computations cannot be modified inplace, using the Embedding.weight tensor before calling Embedding's forward method requires cloning it when max_norm is not None. For example::"

Reply:
I updated this to be similar to your suggestion, just with a couple extra details